### PR TITLE
trigger-control: unstable-2023-06-18 -> 1.5.1

### PR DESCRIPTION
**Pull request by azuwis**
## Description of changes

Changelog https://github.com/Etaash-mathamsetty/trigger-control/releases

Also fix build on darwin/clang, ZHF https://github.com/NixOS/nixpkgs/issues/265948

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tr/trigger-control/package.nix [hunk 1]
```diff
@@ -66,4 +70,4 @@ stdenv.mkDerivation rec {
     maintainers = with maintainers; [ azuwis ];
     platforms = platforms.all;
```
Comment (kirillrdy) [inline pkgs/by-name/tr/trigger-control/package.nix:71]: `mainProgram = "trigger-control";`

Comment (azuwis) [reply pkgs/by-name/tr/trigger-control/package.nix:80]: Done

File: pkgs/by-name/tr/trigger-control/package.nix [hunk 2]
```diff
@@ -41,6 +43,8 @@ stdenv.mkDerivation rec {
     libdecor
   ];
 
+  patches = [ ./clang.patch ];
```
[the remark this answers is not in the document]
Comment (azuwis) [reply pkgs/by-name/tr/trigger-control/package.nix]: Done

File: pkgs/by-name/tr/trigger-control/package.nix [hunk 3]
```diff
@@ -41,6 +44,15 @@ stdenv.mkDerivation rec {
     libdecor
   ];
 
+  patches = [
+    # Fix build on clang https://github.com/Etaash-mathamsetty/trigger-control/pull/23
+    (fetchpatch {
+      name = "clang.patch";
+      url = "https://github.com/Etaash-mathamsetty/trigger-control/pull/23/commits/bbec33296fdac7e2ca0398ae19ca8de8ad883407.patch";
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/tr/trigger-control/package.nix:51]: thanks!
I think the url needs to be:
https://github.com/Etaash-mathamsetty/trigger-control/commit/bbec33296fdac7e2ca0398ae19ca8de8ad883407.patch
as that follows the other patches in nixpkgs, i do not know if the url with  `pull/32/commits` is stable.

Comment (azuwis) [reply pkgs/by-name/tr/trigger-control/package.nix]: Done

File: pkgs/by-name/tr/trigger-control/package.nix [hunk 4]
```diff
@@ -16,15 +17,17 @@ let
...
 
   src = fetchFromGitHub {
     owner = "Etaash-mathamsetty";
     repo = "trigger-control";
-    rev = "d457ebd9e0844cfc456bfa4fa4bb694bb8ad982a";
-    hash = "sha256-QWhUQ8xqS8oRVF0KUpEthlrOoXmhcfEkIHauDI1/5a8=";
+    # upstream does not use consistant tags pattern, so we use git commit hash
+    # https://github.com/Etaash-mathamsetty/trigger-control/tags
+    rev = "7b46e743227830d3a97720067d0a6cf20133af90";
+    hash = "sha256-wueNjrCPWDw7iCOuXgzWZiweRc3o6SUw3xZGgm078nM=";
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/tr/trigger-control/package.nix:30]: I am getting a hash failure -- setting `hash = "";` then running build can get the new hash which is ...
```
hash = "sha256-nWSvsgksZ4Cxy1+i0xy8pNalgsiAuaqxNVwT/CThaBI=";
```

Comment (azuwis) [reply pkgs/by-name/tr/trigger-control/package.nix]: Fixed.

My bad. I way using `rev = "v.1.5.1"` to test the build, and then switched to `rev = "7b46e743227830d3a97720067d0a6cf20133af90"`, without noticing that `fetchFromGitHub` uses different ways to get the source code by tags and commit hash.

Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/tr/trigger-control/package.nix]: 7b46e743227830d3a97720067d0a6cf20133af90 is not the commit referenced by the "v.1.5.1" tag, it is the head on main. so you were using a different rev, 011d0c3657e229c1902cde955e71a64b0c7d91b1 aka "v.1.5.1", which should give the same hash.

anyway, 7b46e743227830d3a97720067d0a6cf20133af90 seems fine as that just makes the version in the header file "1.5.1".
https://github.com/Etaash-mathamsetty/trigger-control/compare/v.1.5.1...main

**Review by ghost — APPROVED** [review]
LGTM. i don't have a controller to test but change looks fine.

Result of `nixpkgs-review pr 271340` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trigger-control</li>
  </ul>
</details>

<hr>

Result of `nixpkgs-review pr 271340` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trigger-control</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**